daliuge-translator/dlg/dropmake/dm_utils.py

```diff
@@ -483,18 +482,15 @@ def convert_construct(lgo):
                 has_app = ak
                 break
         if has_app is None:
-            if node["category"] == ConstructTypes.SUBGRAPH:
-                node["hasInputApp"] = False
             continue
+
         # step 1
         app_node = dict()
         app_node["reprodata"] = node.get("reprodata", {}).copy()
         app_node["key"] = node["key"]
         app_node["category"] = node[has_app]  # node['application']
-        if has_app[0] == "i":
-            app_node["name"] = node["text"] if "text" in node else node["name"]
-        else:
-            app_node["name"] = node["text"] if "text" in node else node["name"]
+        app_node["name"] = node["text"] if "text" in node else node['name']
```

Comment:
I've edited this because the `if/else` clause leads to the exact same result. At some point, these must have been different but have since converged.